functions/packages.bash

```diff
@@ -798,3 +798,160 @@ setup_evcc() {
+  local removeText="This will remove ESPhome dashboard"
+  
+  if [[ $1 == "remove" ]]; then
+    if [[ -n $INTERACTIVE ]]; then
+      whiptail --title "ESPhome dashboard removal" --msgbox "$removeText" 7 80
+    fi
+    echo -n "$(timestamp) [openHABian] Starting ESPHome Dashboard uninstallation..."
+
+    # Function to check the success of the last executed command
+    check_success() {
+        if [ $? -ne 0 ]; then
+            echo -n "$(timestamp) [openHABian] Error: $1"
+            exit 1
+        fi
+    }
+
+    # Ensure the script is run with sudo
+    if [ "$EUID" -ne 0 ]; then
+        echo -n "$(timestamp) [openHABian] Please run this script as root or with sudo."
+        exit 1
+    fi
+
+    # Stop the ESPHome Dashboard service
+    echo -n "$(timestamp) [openHABian] Stopping the ESPHome Dashboard service..."
+    systemctl stop esphome-dashboard.service
+    check_success "Failed to stop ESPHome Dashboard service."
+
+    # Disable the ESPHome Dashboard service
+    echo -n "$(timestamp) [openHABian] Disabling the ESPHome Dashboard service..."
+    systemctl disable esphome-dashboard.service
+    check_success "Failed to disable ESPHome Dashboard service."
+
+    # Remove the ESPHome Dashboard service file
+    echo -n "$(timestamp) [openHABian] Removing the ESPHome Dashboard systemd service file..."
+    rm -f /etc/systemd/system/esphome-dashboard.service
+    check_success "Failed to remove systemd service file."
+
+    # Reload systemd daemon
+    echo -n "$(timestamp) [openHABian] Reloading systemd daemon..."
+    systemctl daemon-reload
+    check_success "Failed to reload systemd daemon."
+
+    # Remove the ESPHome installation directory
+    ESPHOME_DIR="/opt/esphomedashboard"
+    echo -n "$(timestamp) [openHABian] Removing ESPHome directory at $ESPHOME_DIR..."
+    rm -rf "$ESPHOME_DIR"
+    check_success "Failed to remove $ESPHOME_DIR."
+
+    # Final cleanup 
+    echo -n "$(timestamp) [openHABian] Uninstallation complete!"
+    echo -n "$(timestamp) [openHABian] ESPHome Dashboard has been completely removed from the system."
+    return;
+  fi
+
+  if [[ $1 != "install" ]]; then 
+    if [[ -n $INTERACTIVE ]]; then
+      whiptail --title "ESPhome dashboard installation" --msgbox "$installText" 8 80
+    fi
+    echo -n "$(timestamp) [openHABian] Starting ESPHome Dashboard setup..."
+
+    # Function to check the success of the last executed command
+    check_success() {
+        if [ $? -ne 0 ]; then
+            echo -n "$(timestamp) [openHABian] Error: $1"
+            exit 1
+        fi
+    }
+
+    # Ensure the script is run with sudo
+    if [ "$EUID" -ne 0 ]; then
+        echo -n "$(timestamp) [openHABian] Please run this script as root or with sudo."
+        exit 1
+    fi
+
+    # Update system packages
+    echo -n "$(timestamp) [openHABian] Updating system packages..."
+    apt update && apt upgrade -y
+    check_success "Failed to update system packages."
```

Comment:
You can remove this, we already perform this in the background as part of the main script. Additionally, we don't want to upgrade all the packages without user consent as it can cause breaking changes.

Reply:
Solved